daisy/step_wait_for_instances_signal.go

```diff
@@ -150,7 +151,20 @@ func waitForSerialOutput(s *Step, project, zone, name string, so *SerialOutput,
 				return Errf("WaitForInstancesSignal: instance %q: error getting serial port: %v", name, err)
 			}
 			start = resp.Next
-			for _, ln := range strings.Split(resp.Contents, "\n") {
+			lines := strings.Split(resp.Contents, "\n")
+			for i, ln := range lines {
+				// If there is a unconsumed tail string from the previous block of content, concat it with the 1st line of the new block of content.
+				if i == 0 && tailString != "" {
+					ln = tailString + ln
+					tailString = ""
+				}
+
+				// If the content is not ended with a "\n", we want to store the last line as tail string, so it can be concat with the next block of content.
```

Comment:
Where is the check that the last line doesn't end with "\n" performed? This `if` block seems to always store last line as `tailString`.

Reply:
The lines[n] is from spliting by "\n".If there is a "\n" at the end, there will be always a blank entry as the last one. I tested that. So we jsut check ln == ""